src/kernel/boot.c

```diff
@@ -623,16 +623,14 @@ BOOT_CODE static bool_t create_untypeds_for_region(
 }
 
 BOOT_CODE bool_t create_untypeds(cap_t root_cnode_cap,
-                                 region_t boot_mem_reuse_reg)
+                                 p_region_t boot_mem_reuse_reg)
 {
     seL4_SlotPos first_untyped_slot = ndks_boot.slot_pos_cur;
 
     paddr_t start = 0;
     for (word_t i = 0; i < ndks_boot.resv_count; i++) {
         if (start < ndks_boot.reserved[i].start) {
-            region_t reg = paddr_to_pptr_reg((p_region_t) {
-                start, ndks_boot.reserved[i].start
-            });
+            p_region_t reg = (p_region_t) {start, ndks_boot.reserved[i].start};
```

Comment:
There should be no need for the cast `(p_region_t)` there. Using explicit field initializers is good practice. 
```suggestion
            p_region_t reg = {.start = start, .end = ndks_boot.reserved[i].start};
```